### PR TITLE
Fixup automatically created feed permissions

### PR DESCRIPTION
**Pull request by mmitche**
Tighten up the feed permissions used when creating new feeds on the fly, or when publishing signed assets, and do a bit of refactoring in the process.
- Put internal feeds into the internal project, rather than the organization. This is where they should have been from the beginning. There was mainly confusion about project vs. org scoped feeds when this was created.
- Tweak the default feed permissions so that:
  - Members of dnceng get read perms on the local view (this avoids an issue where we used to have to add people as Contributors). This is done by patching the local feed view after feed creation.
  - Admins get ownership
  - the internal and collection build service accounts get write access
- Added some nice NYI exceptions that should fire if we ever change the organization or project that we are running in.
- Update references to the task, in PublishSignedAssets.proj and SetupTargetFeedConfigV3, to use the new task parameters.

Refactoring:
- Remove SetupTargetFeeds.proj (v2 support), which is unused. I chose to do this because it also contain a reference to CreateAzureDevOpsFeed
- Set the default publishing version to 3
- Remove IsInternal as a feed task creation parameter and replace it with the AzDO project. The AzDO project defines the visibility, and this cleans up a little of the dnceng-isms and enables moving them into some helper methods
- Add helper methods for determining some aspects of the feed names
- Refactor some variable names to make them clearer

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/core-eng/tree/master/Documentation/Validation

### CODE REVIEW

**Comment by mmitche** [comment]
This resolves https://github.com/dotnet/core-eng/issues/14599

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/CreateAzureDevOpsFeed.cs [hunk 1]
```diff
@@ -118,38 +132,61 @@ private async Task<bool> ExecuteAsync()
...
-                        if (response.StatusCode == HttpStatusCode.Created)
+                        if (createFeedResponse.StatusCode == HttpStatusCode.Created)
                         {
                             needsUniqueName = false;
                             baseFeedName = versionedFeedName;
+
+                            // Now update the 'Local' feed view with aad tenant visibility.
+                            var feedViewVisibilityPatch = new FeedView()
+                            {
+                                Visibility = "collection"
```
[the remark this answers is not in the document]
Comment (adiaaida) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/CreateAzureDevOpsFeed.cs:151]: Are there ever going to be other visibilities? Is there a finite number of them? Should this be an enum?

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/CreateAzureDevOpsFeed.cs [hunk 2]
```diff
@@ -118,38 +132,61 @@ private async Task<bool> ExecuteAsync()
...
-                        else if (response.StatusCode == HttpStatusCode.Conflict)
+                        else if (createFeedResponse.StatusCode == HttpStatusCode.Conflict)
                         {
                             versionedFeedName = $"{baseFeedName}-{++subVersion}";
                             needsUniqueName = true;
 
                             if (versionedFeedName.Length > MaxLengthForAzDoFeedNames)
                             {
-                                Log.LogError($"The name of the new feed ({baseFeedName}) exceeds the maximum feed name size of 64 chars. Aborting feed creation.");
+                                Console.WriteLine($"The name of the new feed ({baseFeedName}) exceeds the maximum feed name size of 64 chars. Aborting feed creation.");
```
[the remark this answers is not in the document]
Comment (mmitche) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/CreateAzureDevOpsFeed.cs:178]: Ya I had pulled out the task to use in a console app to mess with...this was left over. Good catch.

**Review by adiaaida — APPROVED** [review]
Just some nits

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/CreateAzureDevOpsFeed.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmitche) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/CreateAzureDevOpsFeed.cs:151]: Pulled this out into a parameter instead

[the remark this answers is not in the document]
Comment (mmitche) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/CreateAzureDevOpsFeed.cs]: > Are there ever going to be other visibilities? Is there a finite number of them? Should this be an enum?

There is aadTenant and private. Probably should be an enum. We probably won't use others but worth an enum still.

**Pull request merged**